lighty-core/lighty-codecs/src/main/java/io/lighty/codecs/SerializeIdentifierCodec.java

```diff
@@ -35,6 +42,18 @@ public SerializeIdentifierCodec(final SchemaContext schemaContext) {
         this.dataSchemaContextTree = DataSchemaContextTree.from(schemaContext);
     }
 
+    /**
+     * Returns an unique {@link YangInstanceIdentifier} based on the provided String identifier.
+     * @param identifier URI like String identifier of a node in the schema tree formatted like:
+     *                   <br>
+     *                   {@code MODULE_NAME@REVISION:path_to_node/..}
+     *                   <br>
+     *                   If the specified revision is not present (or is wrongly formatted) and there are
+     *                   multiple revisions of the MODULE_NAME loaded, we perform the serialization on
+     *                   the latest revision.
+     *
+     * @return YangInstanceIdentifier of a node
+     */
     public YangInstanceIdentifier serialize(final String identifier) {
```

Comment:
This method is quite big and complex. Please, try to refactor it into smaller meaningful pieces and ideally add comments.